lua/libcairo_text_helper.h

```diff
@@ -0,0 +1,281 @@
+  for (int i=0; i < glyph_count; ++i) {
+      string_width_in_pixels += glyph_pos[i].x_advance/64;
+  }
+  int draw_x = x;
+
+  if (HB_DIRECTION_IS_VERTICAL(text_direction)) { 
+    /* FIXME */
+    draw_x = width/2 - string_width_in_pixels/2 + x;
+  }
+  else {
+    if (alignment == CAIRO_TEXT_ALIGN_LEFT) {
+      if (text_direction == HB_DIRECTION_RTL) {
+        draw_x = width - x - string_width_in_pixels;
+      }
+      // LTR handled as default.
+    }
+    else if (alignment == CAIRO_TEXT_ALIGN_RIGHT) {
+      if (text_direction == HB_DIRECTION_RTL) {
+        draw_x = x;
+      }
+      else {
+        draw_x = width - x - string_width_in_pixels;
+      }
+    }
+    else if (alignment == CAIRO_TEXT_ALIGN_CENTER) {
+      draw_x = width/2 - string_width_in_pixels/2 + x;
+    }
+  }
+
+  x = draw_x;
+  for (int i=0; i < glyph_count; ++i) {
+      cairo_glyphs[i].index = glyph_info[i].codepoint;
+      cairo_glyphs[i].x = x + (glyph_pos[i].x_offset/64.0);
+      cairo_glyphs[i].y = y - (glyph_pos[i].y_offset/64.0);
+      x += glyph_pos[i].x_advance/64.0;
+      y -= glyph_pos[i].y_advance/64.0;
+  }
+  
+  cairo_set_source_rgba (cr, 1.0, 1.0, 1.0, 1.0);
+  cairo_set_font_face(cr, font->cairo_ft_face);
+  cairo_set_font_size(cr, font->font_size);
+  cairo_show_glyphs(cr, cairo_glyphs, glyph_count);
+
+  free(cairo_glyphs);
+  hb_buffer_destroy(buf);
+}
+
+void cairo_text_hp_simple_show(cairo_t *cr, int x, int y, const char *text, FontData *font)
+{
+  cairo_text_hp_intl_show(cr, x, y, CAIRO_TEXT_ALIGN_LEFT, text, font, "LTR", "Zyyy", "en");
+}
+
+void cairo_text_hp_simple_show_center(cairo_t *cr, int x, int y, const char *text, FontData *font)
+{
+  cairo_text_hp_intl_show(cr, x, y, CAIRO_TEXT_ALIGN_CENTER, text, font, "LTR", "Zyyy", "en");
+}
+
+void cairo_text_hp_simple_show_right(cairo_t *cr, int x, int y, const char *text, FontData *font)
+{
+  cairo_text_hp_intl_show(cr, x, y, CAIRO_TEXT_ALIGN_RIGHT, text, font, "LTR", "Zyyy", "en");
+}
+
+int cairo_text_hp_text_width(const char *text, FontData *font, 
+                             const char *direction, const char *script, const char *language)
+{
+ /* FIXME: Support others */
+  hb_direction_t text_direction = hb_direction_from_string(direction, -1);
+  if (text_direction == HB_DIRECTION_INVALID) {
+    text_direction = HB_DIRECTION_LTR;
+  }
+  hb_script_t text_script = hb_script_from_string(script, -1);
+  if (text_script == HB_SCRIPT_UNKNOWN) {
+    text_script = HB_SCRIPT_COMMON;
+  }
+    
+  hb_language_t text_language = hb_language_from_string (language, -1);
+  
+  /* Draw text */
+  /* Create a buffer for harfbuzz to use */
+  hb_buffer_t *buf = hb_buffer_create();
+
+  //alternatively you can use hb_buffer_set_unicode_funcs(buf, hb_glib_get_unicode_funcs());
+  hb_buffer_set_unicode_funcs(buf, hb_glib_get_unicode_funcs());
+  hb_buffer_set_direction(buf, text_direction); /* or LTR */
+  hb_buffer_set_script(buf, text_script); /* see hb-unicode.h */
```

Comment:
This should be `hb-common.h` as well. Just pointing it out (so it's easier for others to pick up), not that these comments will affect anything.